src/drivers/intel/cavs/timer.c

```diff
@@ -38,11 +38,16 @@ int platform_timer_set(struct timer *timer, uint64_t ticks)
 	if (ticks == 0)
 		ticks = 1;
 
-	/* set new value and run */
-	shim_write64(SHIM_DSPWCT0C, ticks);
+	/* Check if requested time is not past time */
+	if (ticks > shim_read64(SHIM_DSPWC))
+		shim_write64(SHIM_DSPWCT0C, ticks);
+	else
+		shim_write64(SHIM_DSPWCT0C, shim_read64(SHIM_DSPWC) + 1);
+
+	/* Enable IRQ */
 	shim_write(SHIM_DSPWCTCS, SHIM_DSPWCTCS_T0A);
 
-	return 0;
+	return shim_read64(SHIM_DSPWCT0C) - ticks;
```

Comment:
The function returns int and here we can have up to 64 bit value.

Reply:
You suggest to return the 64 bit value which was set I assume. I was also thinking about that but the return of  difference, or 0 if there is no diff would be aliggned better with other .set_timer() interfaces like for BYT platform for example. So the ABI is more a like. What do you think?